pkg/networkpolicy/multinetworkpolicy.go

```diff
@@ -0,0 +1,328 @@
+package networkpolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/k8snetworkplumbingwg/multi-networkpolicy/pkg/apis/k8s.cni.cncf.io/v1beta1"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// MultiNetworkPolicyBuilder provides struct for MultiNetworkPolicy object.
+type MultiNetworkPolicyBuilder struct {
+	// MultiNetworkPolicy definition. Used to create MultiNetworkPolicy object with minimum set of required elements.
+	Definition *v1beta1.MultiNetworkPolicy
+	// Created MultiNetworkPolicy object on the cluster.
+	Object *v1beta1.MultiNetworkPolicy
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before MultiNetworkPolicy object is created.
+	errorMsg string
+}
+
+// NewMultiNetworkPolicyBuilder method creates new instance of builder.
+func NewMultiNetworkPolicyBuilder(apiClient *clients.Settings, name, nsname string) *MultiNetworkPolicyBuilder {
+	glog.V(100).Infof(
+		"Initializing new MultiNetworkPolicyBuilder structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &MultiNetworkPolicyBuilder{
+		apiClient: apiClient,
+		Definition: &v1beta1.MultiNetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the MultiNetworkPolicy is empty")
+
+		builder.errorMsg = "The MultiNetworkPolicy 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the MultiNetworkPolicy is empty")
+
+		builder.errorMsg = "The MultiNetworkPolicy 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// WithPodSelector adds podSelector to MultiNetworkPolicy.
+func (builder *MultiNetworkPolicyBuilder) WithPodSelector(
+	podSelector *metav1.LabelSelector) *MultiNetworkPolicyBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating MultiNetworkPolicy %s in %s namespace with the podSelector defined: %v",
+		builder.Definition.Name, builder.Definition.Namespace, podSelector)
+
+	if podSelector == nil {
+		glog.V(100).Infof("The podSelector can not be empty map")
+
+		builder.errorMsg = "The podSelector is an empty map"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	builder.Definition.Spec.PodSelector = *podSelector
+
+	return builder
+}
+
+// WithAnnotation adds Annotation to the MultiNetworkPolicy.
+func (builder *MultiNetworkPolicyBuilder) WithAnnotation(annotation string) *MultiNetworkPolicyBuilder {
```

Comment:
Personally, I think this method name is a bit misleading. It doesn't accept just any user-defined annotation but only an annotation value with k8s.v1.cni.cncf.io/policy-for as the key.  I think it would be more appropriate to name it something like `WithPolicyForAnnotation` or something similar that you think sufficiently conveys what this method does.

Reply:
Refactored